src/battle/partner/lakilester.c

```diff
@@ -19,7 +44,244 @@ ApiStatus func_80238000_70BD10(Evt* script, s32 isInitialCall) {
+                partnerState->currentPos.y += partnerState->speed * cosTheta;
+            }
+            if (partnerState->currentPos.x < -30.0f) {
+                partnerState->currentPos.x = -30.0f;
+            }
+            if (partnerState->currentPos.x > 170.0f) {
+                partnerState->currentPos.x = 170.0f;
+            }
+            if (partnerState->currentPos.y > 130.0f) {
+                partnerState->currentPos.y = 130.0f;
+            }
+            if (partnerState->currentPos.y < 0.0f) {
+                partnerState->currentPos.y = 0.0f;
+            }
+            if (battleStatus->currentButtonsPressed & 0x8000) {
+                D_8023D298 = 0;
+            }
+            if (D_8023D298 == 60) {
+                set_animation(0x100, 0, 0x100010);
+            }
+            if (D_8023D298 == 30) {
+                set_animation(0x100, 0, 0x100011);
+            }
+            if (D_8023D298 != 0) {
+                D_8023D298--;
+                break;
+            }
+            script->varTable[14] = 0;
+            script->varTable[15] = 0;
+            script->varTable[7] = partnerState->currentPos.x;
+            script->varTable[8] = partnerState->currentPos.y;
+            script->varTable[9] = partnerState->currentPos.z;
+            partnerState->distance = dist2D(partnerState->currentPos.x, partnerState->currentPos.y, partnerState->goalPos.x, partnerState->goalPos.y) / partnerState->unk_24;
+            if (partnerState->distance <= 12.0) {
+                script->varTable[15] = 1;
+            }
+            hud_element_free(D_8023D28C);
+            hud_element_free(D_8023D288);
+            hud_element_free(D_8023D2BC);
+            for (i = 0; i < 1; i++) {
+                id = D_8023D2B8[i];
+                hud_element_free(id);
+            }
+            btl_set_popup_duration(0);
+            sfx_stop_sound(0x312);
+            return ApiStatus_DONE2;
+    }
+
+    get_screen_coords(gCurrentCameraID,
+                      partnerState->goalPos.x, partnerState->goalPos.y, partnerState->goalPos.z,
+                      &screenX, &screenY, &sp28);
+    id = D_8023D28C;
+    hud_element_set_render_pos(id, screenX, screenY);
+    hud_element_set_transform_rotation(id, 0.0f, 0.0f, D_8023D290);
+    D_8023D290 -= 10;
+    D_8023D290 = clamp_angle(D_8023D290);
+    get_screen_coords(gCurrentCameraID,
+                      partnerState->currentPos.x, partnerState->currentPos.y, partnerState->currentPos.z,
+                      &screenX, &screenY, &sp28);
+    hud_element_set_render_pos(D_8023D288, screenX, screenY);
+
+    temp_v1_2 = script->functionTemp[0];
+    if (temp_v1_2 < 3 && temp_v1_2 > 0) {
+        D_8023D2C0 += 20;
+        if (D_8023D2C0 > 50) {
+            D_8023D2C0 = 50;
+        }
+        hud_element_set_render_pos(D_8023D2BC, D_8023D2C0, D_8023D2C4);
+    }
+
+    temp_a0_2 = script->functionTemp[0];
+    if (temp_a0_2 < 3) {
```

Comment:
can these be combined, as a few lines above?

Reply:
this section is part of why the function doesn't match - I'd prefer to leave it as-is for now until we figure out how to match it

Reply:
combining didn't help when I was trying to match it earlier